component/portal/src/main/java/org/exoplatform/portal/rest/GroupRestResourcesV1.java

```diff
@@ -641,25 +646,37 @@ public Response getGroupsTreeMembers(@Context
 
     offset = offset > 0 ? offset : DEFAULT_OFFSET;
     limit = limit > 0 ? limit : DEFAULT_LIMIT;
-
-    Collection<Group> UserGroupsList = organizationService.getGroupHandler()
-                                                          .findGroupsOfUserByKeyword(identity.getUserId(), q, groupType);
-    int totalSize = UserGroupsList.size();
-    int limitToFetch = limit;
-    if (totalSize < (offset + limitToFetch)) {
-      limitToFetch = totalSize - offset;
-    }
+    CollectionEntity<Group> result = null;
```

Comment:
What is the initial aim of this endpoint treeMembers ?

I'm afraid that we are changing the purpose of this endpoint to add a special case for administrators.

Reply:
This endpoint was added for process development to add groups to the suggester, I don't think that it can be used otherwhere. but we will add a new query param to this endpoint to not change it behavior if no need to get all groups for admins.